src/libraries/System.Diagnostics.Process/tests/ProcessThreadTests.cs

```diff
@@ -148,17 +149,22 @@ public async Task TestStartTimeProperty()
                 await Task.Factory.StartNew(() =>
                 {
                     p.Refresh();
-                    try
-                    {
-                        var newest = p.Threads.Cast<ProcessThread>().OrderBy(t => t.StartTime.ToUniversalTime()).Last();
-                        Assert.InRange(newest.StartTime.ToUniversalTime(), curTime - allowedWindow, DateTime.Now.ToUniversalTime() + allowedWindow);
-                    }
-                    catch (InvalidOperationException)
-                    {
-                        // A thread may have gone away between our getting its info and attempting to access its StartTime
-                    }
+
+                    int newThreadId = OperatingSystem.IsWindows() ? GetCurrentThreadId() : gettid();
+
+                    ProcessThread[] processThreads = p.Threads.Cast<ProcessThread>().ToArray();
+                    ProcessThread newThread = Assert.Single(processThreads, thread => thread.Id == newThreadId);
+
+                    Assert.InRange(newThread.StartTime.ToUniversalTime(), curTime - allowedWindow, DateTime.Now.ToUniversalTime() + allowedWindow);
                 }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
             }
+
+            [DllImport("libc")]
+            static extern int gettid();
```

Comment:
[CI failures](https://[helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-104972-merge-91f33105ce7d418cab/System.Diagnostics.Process.Tests/1/console.e209f89d.log?helixlogtype=result](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-104972-merge-91f33105ce7d418cab/System.Diagnostics.Process.Tests/1/console.e209f89d.log?helixlogtype=result)) are related to this method. `System.EntryPointNotFoundException : gettid`.